docs/topics/dokka-migration.md

```diff
@@ -77,157 +95,411 @@ After updating the Dokka Gradle plugin to v2, follow the migration steps applica
 DGP v2 introduces some changes in the [Gradle configuration options](dokka-gradle.md#configuration-options). In the `build.gradle.kts` file, adjust the configuration 
 options according to your project setup: 
 
-* **New top-level DSL configuration:** Replace the old configuration syntax with the 
-  new top-level `dokka {}` DSL configuration. For example:
-
-  Previous configuration:
-
-  ```kotlin
-  tasks.withType<DokkaTask>().configureEach {
-      dokkaSourceSets {
-          named("main") {
-              moduleName.set("Project Name")
-              includes.from("README.md")
-              sourceLink {
-                  localDirectory.set(file("src/main/kotlin"))
-                  remoteUrl.set(URL("https://example.com/src"))
-                  remoteLineSuffix.set("#L")
-              }
-          }
-      }
-  }
-
-  tasks.dokkaHtml {
-      pluginConfiguration<DokkaBase, DokkaBaseConfiguration> {
-          customStyleSheets.set(listOf("styles.css"))
-          customAssets.set(listOf("logo.png"))
-          footerMessage.set("(c) Your Company")
-      }
-  }
-  ```
- 
-  New configuration:
-
-  ```kotlin
-  dokka {
-      moduleName.set("Project Name")
-      dokkaSourceSets.main {
-          includes.from("README.md")
-          sourceLink {
-              localDirectory.set(file("src/main/kotlin"))
-              remoteUrl("https://example.com/src")
-              remoteLineSuffix.set("#L")
-          }
-      }
-      pluginsConfiguration.html {
-          customStyleSheets.from("styles.css")
-          customAssets.from("logo.png")
-          footerMessage.set("(c) Your Company")
-      }
-  }
-  ```
-
-* **Visibility settings:**  The `documentedVisibilities` property has changed from `Visibility.PUBLIC` to `VisibilityModifier.Public`.
-
-  Previous configuration:
-
-  ```kotlin
-  import org.jetbrains.dokka.DokkaConfiguration.Visibility
-  // ...
-  documentedVisibilities.set(
-      setOf(Visibility.PUBLIC)
-  )  
-  ```
+#### New top-level DSL configuration
 
-  New configuration:
+Replace the old configuration syntax with the new top-level `dokka {}` DSL configuration. For example:
 
-  ```kotlin
-  import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
-  // ...
-  documentedVisibilities.set(
-      setOf(VisibilityModifier.Public)
-  )
-  
-  // OR
-  
-  documentedVisibilities(VisibilityModifier.Public)
-  ```
+Previous configuration:
+
+```kotlin
+tasks.withType<DokkaTask>().configureEach {
```

Comment:
@AlejandraPedroza, @adam-enko 
Inspired by #4026, I think that it would be nice to also show here that some properties are moved from root configuration on task to `dokkaPublications.html` (like we show explicitly below for `outputDirectory`)

Currently, it's almost impossible to guess, that we split properties in two places and not just removed them.